SwiftUI/Model/SearchOperation/SearchOperation.mm

```diff
@@ -49,29 +49,31 @@ - (id)initWithSearchText:(NSString *)searchText zimFileIDs:(NSSet *)zimFileIDs {
 
 /// Perform index and title based searches.
 - (void)performSearch {
-    // get a list of archives that are included in search
-    typedef std::unordered_map<std::string, zim::Archive> archives_map;
-    auto *allArchives = static_cast<archives_map *>([[ZimFileService sharedInstance] getArchives]);
-    
-    std::vector<zim::Archive> indexSearchArchives = std::vector<zim::Archive>();
-    std::vector<zim::Archive> titleSearchArchives = std::vector<zim::Archive>();
-    for (NSUUID *zimFileID in self.zimFileIDs) {
-        std::string zimFileID_C = [[[zimFileID UUIDString] lowercaseString] cStringUsingEncoding:NSUTF8StringEncoding];
-        try {
-            auto archive = allArchives->at(zimFileID_C);
-            if (archive.hasFulltextIndex()) {
-                indexSearchArchives.push_back(archive);
-            }
-            titleSearchArchives.push_back(archive);
-        } catch (std::out_of_range) { }
-    }
-    
-    // perform index and title search
-    [self addIndexSearchResults:indexSearchArchives];
-    if (titleSearchArchives.size() > 0) {
-        int count = std::max((35 - (int)[self.results count]) / (int)titleSearchArchives.size(), 5);
-        [self addTitleSearchResults:titleSearchArchives count:(int)count];
-    }
+    try {
+        // get a list of archives that are included in search
+        typedef std::unordered_map<std::string, zim::Archive> archives_map;
+        auto *allArchives = static_cast<archives_map *>([[ZimFileService sharedInstance] getArchives]);
+
+        std::vector<zim::Archive> indexSearchArchives = std::vector<zim::Archive>();
+        std::vector<zim::Archive> titleSearchArchives = std::vector<zim::Archive>();
+        for (NSUUID *zimFileID in self.zimFileIDs) {
+            std::string zimFileID_C = [[[zimFileID UUIDString] lowercaseString] cStringUsingEncoding:NSUTF8StringEncoding];
+            try {
+                auto archive = allArchives->at(zimFileID_C);
+                if (archive.hasFulltextIndex()) {
+                    indexSearchArchives.push_back(archive);
+                }
+                titleSearchArchives.push_back(archive);
+            } catch (std::out_of_range) { }
+        }
+
+        // perform index and title search
+        [self addIndexSearchResults:indexSearchArchives];
+        if (titleSearchArchives.size() > 0) {
+            int count = std::max((35 - (int)[self.results count]) / (int)titleSearchArchives.size(), 5);
+            [self addTitleSearchResults:titleSearchArchives count:(int)count];
+        }
+    }  catch (std::exception) {}
```

Comment:
Sadly, xapian exception are not inheriting standard exception.
So you should also catch [Xapian::Error](https://xapian.org/docs/apidoc/html/classXapian_1_1Error.html)

Reply:
Once you have fixed libzim, it wil!?

Reply:
This should be handled in https://github.com/openzim/libzim/issues/870

Reply:
Do you think we could stick to the convention of throwing only `std::exception`? 
If  `Xapian::Error` could be caught by the libzim, and re-thrown as `std::exception` that would be the simplest / cleanest solution for client apps on Apple.

Reply:
Yes IMHO

Reply:
> Do you think we could stick to the convention of throwing only std::exception?
If Xapian::Error could be caught by the libzim, and re-thrown as std::exception that would be the simplest / cleanest solution for client apps on Apple.

Yes, that's the plan with https://github.com/openzim/libzim/pull/873, we will throw a `zim::zim::ZimFileFormatError` (which inherit `std::exception`) or handle the exception internally to returns default value.
When we will do one or the other solution is still open to discussion (and evolution), but it will always be one of the two ways (or a bug in libzim code).